File: backend/internal/domain/models/scoring.go
package models

type ScoringResult struct {
	PredictIncome   float64            `json:"predict_income"`
	Recommendations []string           `json:"recommendations"`
	Factors         map[string]float64 `json:"factors"`
	PositiveFactors []string           `json:"positive_factors"`
	NegativeFactors []string           `json:"negative_factors"`
	CreditLimit     float64            `json:"credit_limit"`
	MaxCreditLimit  float64            `json:"max_credit_limit"`
}

func (s *ScoringResult) IsValid() bool {
	return s.PredictIncome >= 0
}

File: frontend/src/app/core/models/scoring.ts
export interface Scoring {
  id: number;
  first_name: string;
  last_name: string;
  middle_name?: string;
  birth_date: string;
  income?: number;
  predict_income: number;
  credit_limit: number;
  max_credit_limit?: number;
  recommendations: string[];
  positive_factors: string[];
  negative_factors: string[];
}
